### PR TITLE
12323 - Add Beanshell Range functions

### PR DESCRIPTION
**Pull request by BrentEaston**
Built on the FastRangeV2 branch. Adds a variety of Range Beanshell functions

### CODE REVIEW

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+ RangeInPixels(attachmentName)
+ RangeInCells(attachmentName)
+
+Calculate the range from the current piece to the position of the first piece attached to this piece using the Attachment Name "attachmentName" that is also on the same map as the current piece.
+
+ RangeInPixels(x1, y1, x2, y2)
+ RangeInCells(x1, y1, x2, y2)
+
+Calculate the range from the position x2, y1 to x2, y2 on the current map
```
Comment (jonathanrwatts) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc:36]: First "x2" should be "x1"

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+ RangeInPixels(x1, y1, x2, y2, mapName)
+ RangeInCells(x1, y1, x2, y2, mapName)
+
+Calculate the range from the position x2, y1 to x2, y2 on the named map. This is the only form of the Range functions that can be used outside of piece traits.
```
Comment (jonathanrwatts) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc:41]: First "x2" should be "x1"

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc (hunk 1 above)
Comment (BrentEaston) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc]: Fixed

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc (hunk 2 above)
Comment (BrentEaston) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionRange.adoc]: Fixed

**Comment by uckelman** [comment]
Rebased over main as #12380, merged there.

**Pull request closed without merging**